### PR TITLE
DOCS audit attributes for API Key authn

### PR DESCRIPTION
**Pull request by albertzaharovits**
This PR describes the new audit entry attributes `api_key.id`, `api_key.name` and `authentication.type`, as well as the meaning of existing attributes when authentication is performed using API keys.

This should've been part of https://github.com/elastic/elasticsearch/pull/58928

### CODE REVIEW

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 1]
```diff
@@ -157,16 +157,35 @@ There are a few events that have some more attributes in addition to those
 that have been previously described:
 
 * `authentication_success`:
-  `realm`              ::   The name of the realm that successfully
-                            authenticated the user.
+  `realm`              ::   The name of the realm that successfully authenticated the user,
+                            or the special value of `_es_api_key` if authentication is
+                            performed using an API Key. This is a shorthand attribute
+                            for the same information that is described by the `user.realm`,
+                            `user.run_by.realm` and `authentication.type` attributes.
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:164]:
```suggestion
  `realm`              ::   The name of the realm that successfully authenticated the user.
                            If authenticated using an API key, this is the special value of
                            `_es_api_key`. This is a shorthand attribute
                            for the same information that is described by the `user.realm`,
                            `user.run_by.realm` and `authentication.type` attributes.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 2]
```diff
@@ -157,16 +157,35 @@ There are a few events that have some more attributes in addition to those
...
   `user.name`          ::   The name of the _effective_ user. This is usually the
                             same as the _authenticated_ user, but if using the
                             <<run-as-privilege, run as authorization functionality>>
                             this instead denotes the name of the  _impersonated_ user.
+                            If authentication is performed using an API Key this attribute
+                            designates the name of the owner user of the key.
+  `user.realm`         ::   The realm name that the _effective_ user belongs to. If authentication
+                            is performed using an API Key this attribute designates the realm name
+                            that the owner of the key belongs to.
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:173]:
```suggestion
                            If authenticated using an API key, this is
                            the name of the API key owner.
  `user.realm`         ::   Name of the the realm to which the _effective_ user 
                            belongs. If authenticated using an API key, this is
                            the name of the realm to which the API key owner belongs.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 3]
```diff
@@ -157,16 +157,35 @@ There are a few events that have some more attributes in addition to those
...
+                            using the <<run-as-privilege, run as authorization functionality>>
+                            and denotes the name of the realm that the _authenticated_
+                            (_impersonator_) user belongs to.
+  `authentication.type`::   One of `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
+  `api_key.id`         ::   This attribute is present only if authentication is performed using an
+                            API Key, and contains the value for the key id that is returned in response
+                            to the <<security-api-create-api-key, Create API Key>> call.
+  `api_key.name`       ::   This attribute is present only if authentication is performed using an
+                            API Key, and contains the value for the key name given in the request
+                            to <<security-api-create-api-key, Create API Keys>>
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:188]:
```suggestion
  `user.run_by.realm`  ::   Name of the realm to which the _authenticated_
                            (_impersonator_) user belongs.
                            This attribute is provided only if the request
                            uses the <<run-as-privilege, run as authorization functionality>>.
  `authentication.type`::   Method used to authenticate the user.
                            Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 4]
```diff
@@ -188,13 +207,19 @@ that have been previously described:
...
 
-* `access_granted` or `access_denied`:
-  `user.roles`         ::    The role names of the user as an array.
-  `user.name`          ::    The name of the _effective_ user that is being
-                             authorized or unauthorized. This is usually the _authenticated_
-                             user, but if using the <<run-as-privilege, run as authorization functionality>>
+* `access_granted` and `access_denied`:
+  `user.roles`         ::    The role names of the user as an array. If authentication is
+                             performed using an API Key this attribute designates the
+                             role names of the owner user of the key.
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:213]:
```suggestion
  `user.roles`         ::    The role names of the user as an array. If authenticated
                             using an API key, this contains the
                             role names of the API key owner.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 5]
```diff
@@ -188,13 +207,19 @@ that have been previously described:
...
+  `user.name`          ::    The name of the _effective_ user. This is usually the
+                             same as the _authenticated_ user, but if using the
+                             <<run-as-privilege, run as authorization functionality>>
                              this instead denotes the name of the  _impersonated_ user.
-  `user.realm`         ::    The realm name that the _effective_ user belongs to.
+                             If authentication is performed using an API Key this attribute
+                             designates the name of the owner user of the key.
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:219]:
```suggestion
                             If authenticated using an API key, this is
                             the name of the API key owner.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 6]
```diff
@@ -188,13 +207,19 @@ that have been previously described:
...
+  `user.realm`         ::    The realm name that the _effective_ user belongs to. If authentication
+                             is performed using an API Key this attribute designates the realm name
+                             that the owner of the key belongs to.
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:222]:
```suggestion
  `user.realm`         ::   Name of the the realm to which the _effective_ user 
                            belongs. If authenticated using an API key, this is
                            the name of the realm to which the API key owner belongs.
```

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 7]
```diff
@@ -203,6 +228,13 @@ that have been previously described:
                              using the <<run-as-privilege, run as authorization functionality>>
                              and denotes the name of the realm that the _authenticated_
                              (_impersonator_) user belongs to.
+  `authentication.type`::    One of `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
+  `api_key.id`         ::    This attribute is present only if authentication is performed using an
+                             API Key, and contains the value for the key id that is returned in response
+                             to the <<security-api-create-api-key, Create API Key>> call.
+  `api_key.name`       ::    This attribute is present only if authentication is performed using an
+                             API Key, and contains the value for the key name given in the request
+                             to <<security-api-create-api-key, Create API Keys>>
```
Comment (jrodewig) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:237]:
```suggestion
  `authentication.type`::   Method used to authenticate the user.
                            Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
```

**Review by jrodewig — APPROVED** [review]
LGTM.

Left some non-blocking suggestions. Feel free to take or ignore as you see fit. In general, they split the sentences and ensure that
`API key` is correctly capitalized.

**Comment by albertzaharovits** [comment]
Thank you @jrodewig ! It is much better phrased like this.

**Review by tvernum — APPROVED** [review]
LGTM

**Pull request merged**